src/actions/Button.tsx

```diff
@@ -0,0 +1,125 @@
+
+import "./Button.scss"
+
+export const isExternalLink = (href: string) => {
+  return href.startsWith("http://") || href.startsWith("https://")
+}
+
+export const isInternalLink = (href: string) => {
+  return href.startsWith("/") && !href.startsWith("//")
+}
+
+export interface ButtonProps {
+  /** Button content */
+  children: React.ReactNode
+  /** Appearance of the component */
+  variant?:
+    | "primary"
+    | "primary-outlined"
+    | "secondary"
+    | "secondary-outlined"
+    | "alert"
+    | "alert-outlined"
+    | "highlight" // or accent?
+    | "highlight-outlined" // or accent?
+    | "borderless-text"
+  /** Button size */
+  size?: "sm" | "md" | "lg"
+  /** Icon to show before the label text */
+  leadIcon?: React.ReactNode
+  /** Icon to show after the label text */
+  tailIcon?: React.ReactNode
+  /** Event handler for the button click */
+  onClick?: (e: React.MouseEvent) => void
+  /** Use an `<a href>` tag instead of `<button>` for a hyperlink */
+  href?: string
+  /** HTML button type */
+  type?: "button" | "submit" | "reset"
+  /** Set to true to disable the button */
+  disabled?: boolean
+  /** Set to true to hide the button from he accessibility tree */
+  ariaHidden?: boolean
+  /** Accessible label if button doesn't contain text content */
+  ariaLabel?: string
+  /** Element ID */
+  id?: string
+  /** Additional CSS classes */
+  className?: string
+}
+
+const setupButtonProps = (props: ButtonProps) => {
+  const classNames = ["button"]
+
+  const tailIcon =
+    props.href && isExternalLink(props.href) ? (
+      <Icon icon={faArrowUpRightFromSquare} />
+    ) : (
+      props.tailIcon
+    )
+
+  if (props.className) classNames.push(props.className)
+  if (props.leadIcon) classNames.push("has-lead-icon")
+  if (tailIcon) classNames.push("has-tail-icon")
+
+  return {
+    ...props,
+    "data-variant": props.variant || "primary",
+    "data-size": props.size,
+    className: classNames.join(" "),
+    type: props.type || "button",
+    "aria-label": props.ariaLabel,
+    "aria-hidden": props.ariaHidden,
+    tabIndex: props.ariaHidden ? -1 : null,
+    tailIcon,
+  }
+}
+
+const ButtonElement = (props: ButtonProps) => {
+  return <button {...props} />
+}
+
+const LinkButton = (props: ButtonProps) => {
+  if (props.href && isExternalLink(props.href)) {
+    return (
+      <a href={props.href} target="_blank" {...props}>
+        {props.children}
+      </a>
+    )
+  } else {
+    const { LinkComponent } = useContext(NavigationContext)
+    return (
+      <LinkComponent href={props.href} {...props}>
+        {props.children}
+      </LinkComponent>
+    )
+  }
+}
+
+const Button = (props: ButtonProps) => {
+  const updatedProps = setupButtonProps(props)
+
+  const buttonInner = (
+    <>
+      {updatedProps.leadIcon}
+      {updatedProps.children}
+      {updatedProps.tailIcon}
+    </>
+  )
+
+  ;["leadIcon", "tailIcon", "ariaLabel", "ariaHidden"].forEach((key) => {
```

Comment:
is this semicolon needed?

Reply:
We do need it with this particular syntax, otherwise it causes a JS error because the `[]` gets attached to the preceeding statement. Any suggestion for a refactor most appreciated!

Reply:
I personally am finding this way of building props + then removing them to be a bit confusing, I would prefer doing some of them inline

Reply:
Yeah, maybe it's too clever by half. I can work on a more straightforward approach.